packages/block-editor/src/components/spacing-sizes-control/test/utils.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Internal dependencies
+ */
+import {
+	isValueSpacingPreset,
+	getCustomValueFromPreset,
+	getSpacingPresetCssVar,
+	getSpacingPresetSlug,
+	getSliderValueFromPreset,
+	getAllRawValue,
+	isValuesMixed,
+	isValuesDefined,
+} from '../utils';
+
+describe( 'isValueSpacingPreset', () => {
+	it( 'should return true if value is string in spacing presets var format', () => {
+		expect( isValueSpacingPreset( 'var:preset|spacing|20' ) ).toBe( true );
+	} );
+	it( 'should return false if value is not a string in spacing presets var format', () => {
+		expect( isValueSpacingPreset( '30px' ) ).toBe( false );
+	} );
+} );
+
+describe( 'getCustomValueFromPreset', () => {
+	const spacingSizes = [ { name: 'Small', slug: 20, size: '8px' } ];
+	it( 'should return original value if not a string in spacing presets var format', () => {
+		expect( getCustomValueFromPreset( '20px', spacingSizes ) ).toBe(
+			'20px'
+		);
+	} );
+	it( 'should return value from matching spacingSizes array entry if string in spacing presets var format', () => {
+		expect(
+			getCustomValueFromPreset( 'var:preset|spacing|20', spacingSizes )
+		).toBe( '8px' );
+	} );
+	it( 'should return undefined if no matching preset in spacingSizes array', () => {
+		expect(
+			getCustomValueFromPreset( 'var:preset|spacing|30', spacingSizes )
+		).toBe( undefined );
+	} );
+} );
+
+describe( 'getSpacingPresetCssVar', () => {
+	it( 'should return original value if not a string in spacing presets var format', () => {
+		expect( getSpacingPresetCssVar( '20px' ) ).toBe( '20px' );
+	} );
+	it( 'should return a string in spacing presets css var format with slug from provided value', () => {
+		expect( getSpacingPresetCssVar( 'var:preset|spacing|20' ) ).toBe(
+			'var(--wp--preset--spacing--20)'
+		);
+	} );
+} );
+
+describe( 'getSpacingPresetSlug', () => {
+	it( 'should return original value if 0 or default', () => {
+		expect( getSpacingPresetSlug( '0' ) ).toBe( '0' );
+		expect( getSpacingPresetSlug( 'default' ) ).toBe( 'default' );
+	} );
+	it( 'should return the int value of the slug portion of a valid preset var', () => {
+		expect( getSpacingPresetSlug( 'var:preset|spacing|20' ) ).toBe( 20 );
+	} );
+} );
+
+describe( 'getSliderValueFromPreset', () => {
+	const spacingSizes = [
+		{ name: 'Small', slug: 20, size: '8px' },
+		{ name: 'Large', slug: 30, size: '24px' },
+	];
+	it( 'should return NaN if no matching preset found', () => {
+		expect( getSliderValueFromPreset( '10px', spacingSizes ) ).toBe( NaN );
+	} );
+	it( 'should return the int value of the slug portion of a valid preset var', () => {
+		expect(
+			getSliderValueFromPreset( 'var:preset|spacing|30', spacingSizes )
+		).toBe( 1 );
+	} );
+} );
+
+describe( 'getAllRawValue', () => {
```

Comment:
Just flagging that `mode()` has a bias for last values, so

```js
console.log( mode(['5px',  '5px', '2px', '2px']) ); // 2px
```

Will that matter? And should we test for it?

Reply:
I don't think it matters in this instance